src/components/organisms/AdminVenueView/components/RunTabRooms/RunTabRooms.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useMemo } from "react";
+import { groupBy } from "lodash";
+
+import { Room } from "types/rooms";
+import { AnyVenue, isVenueWithRooms } from "types/venues";
+
+import { WithId } from "utils/id";
+
+import { useVenueEvents } from "hooks/events";
+
+import { RoomCard } from "components/organisms/AdminVenueView/components/RoomCard/RoomCard";
+
+import { ButtonNG } from "components/atoms/ButtonNG/ButtonNG";
+
+import "./RunTabRooms.scss";
+
+export interface RunTabRoomsProps {
+  venue: WithId<AnyVenue>;
+}
+
+export const RunTabRooms: React.FC<RunTabRoomsProps> = ({ venue }) => {
+  const venueWithRooms = isVenueWithRooms(venue);
+  const rooms = venueWithRooms ? venue?.rooms ?? [] : [];
```

Comment:
maybe we should create a const for an empty array, because currently it'll create a new object on every rerender and will cause other recalculations.

Reply:
A constant for an empty array is
```typescript
const EMPTY_ARRAY= Object.freeze([]);
```

otherwise, just sharing a constant reference to a variable opens the possibility of a mutable object being shared across components separated both in space and in time.

Reply:
BTW, how much of an impact to the execution of JS does a creation of an empty array is?